src/app/components/Embeds/OEmbed/index.tsx

```diff
@@ -58,4 +58,4 @@ const OEmbedLoader = ({ oembed }: OEmbedProps) => {
   return <EmbedHtml embeddableContent={html} />;
 };
 
-export default OEmbedLoader;
+export default memo(OEmbedLoader);
```

Comment:
silly q for me really : i think this memo means the embeds on the page won't get re-rendered unless they change, is this change related to the optimizely changes for the OJs ? or have you just seen it and thoughht it's worth doing quickly

Reply:
Its related to the Optimizely integration here as the variation hook causes a re-render which caused the embeds to break. Not _entirely_ sure why tbh, but I feel like memoising the Embeds isn't a bad thing in general as the state of them shouldn't change on our end during renders.

Reply:
Yea not entirely sure whats different between this implementation and previous Optimizely implementations. I do call the `useOptimizelyVariation` hook multiple times in sub-components, but would have expected that to not cause a re-render here since its down a level.

Reply:
It might be the usage at Article page level? https://github.com/bbc/simorgh/blob/cd94163db9f5ab909f5307080db97d409d70a898/src/app/pages/ArticlePage/ArticlePage.tsx#L138-L140

You might be able to encapsulate the state if you use the hook in [getWsojComponent](https://github.com/bbc/simorgh/blob/cd94163db9f5ab909f5307080db97d409d70a898/src/app/pages/ArticlePage/ArticlePage.tsx#L109), and creating a component for [ATIAnalytics](https://github.com/bbc/simorgh/blob/cd94163db9f5ab909f5307080db97d409d70a898/src/app/pages/ArticlePage/ArticlePage.tsx#L254), to call the hook as well. Downside is the hook would be called twice

Reply:
Yea I thought it was this instance as well, but we had it before in the JumpTo and it didn't cause this issue: https://github.com/bbc/simorgh/pull/12373

Maybe calling the `useOptimizelyVariation` hook multiple times causes it trigger an update in the `OptimizelyProvider`, which then re-renders all the children?

Reply:
The problem does seem to stem from the `useOptimizelyVariation` hook itself, more so the `useDecision` hook as it runs twice, once on initial re-render and then again once the Optimizely config has been downloaded and the variation can be determined.

Reply:
The `useDecision` initial return value is:
<img width="476" alt="Screenshot 2025-02-26 at 15 23 46" src="https://github.com/user-attachments/assets/2f71539c-733b-4616-afe8-f1fb3e4ba29a" />

Reply:
Feel like its not great performance wise, but probably ok for experimentation?